### PR TITLE
Visualization updates to support no results page and server side rendering

### PR DESCRIPTION
**Pull request by sarveshnagpal**

### CODE REVIEW

File: src/visualization/components/Session.tsx [hunk 1]
```diff
@@ -7,20 +7,31 @@ import NextIcon from 'material-ui/svg-icons/navigation/chevron-right';
...
-            var active = impression.envelope.impressionId === this.props.impression.envelope.impressionId;
-            var dateTime = new Date(impression.envelope.time);
-            var title = (impression.envelope.title ? impression.envelope.title : this.getHostname(impression.envelope.url));
+            let active = impression.envelope.impressionId === this.props.impression.envelope.impressionId;
+            let dateTime = new Date(impression.envelope.dateTime);
+            let url = impression.envelope.url;
+            let vertical = impression.envelope.vertical ? impression.envelope.vertical : null;
+            let time = dateTime.toLocaleString('en-US', { hour: 'numeric', minute:'numeric', second:'numeric', hour12: false });
+            let title = (impression.envelope.title ? impression.envelope.title : this.getHostname(impression.envelope.url));
+            let header = vertical ? `${time} // ${vertical}` : time;
```
Comment (ender336) [inline src/visualization/components/Session.tsx]: >// ${vertical}` : time; [](start = 45, length = 23)

?

File: src/visualization/components/Session.tsx [hunk 2]
```diff
@@ -7,20 +7,31 @@ import NextIcon from 'material-ui/svg-icons/navigation/chevron-right';
...
     renderSession() {
         return this.props.session.map((impression) => {
-            var active = impression.envelope.impressionId === this.props.impression.envelope.impressionId;
-            var dateTime = new Date(impression.envelope.time);
-            var title = (impression.envelope.title ? impression.envelope.title : this.getHostname(impression.envelope.url));
+            let active = impression.envelope.impressionId === this.props.impression.envelope.impressionId;
+            let dateTime = new Date(impression.envelope.dateTime);
+            let url = impression.envelope.url;
+            let vertical = impression.envelope.vertical ? impression.envelope.vertical : null;
+            let time = dateTime.toLocaleString('en-US', { hour: 'numeric', minute:'numeric', second:'numeric', hour12: false });
```
Comment (ender336) [inline src/visualization/components/Session.tsx]: >time [](start = 16, length = 4)

nit: I think this is more descriptive as displayTime

File: src/visualization/parsers/boxmodel.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    private document: Document;
+    private base: string;
+   
+    setup(document: Document, frame: HTMLIFrameElement, base: string, thumbnail? : boolean) {
+        this.layouts = {};
+        this.document = document;
+        this.base = base;
+    }
+
+    private domInsert(node: Node, parent?: Node, nextSibling?: Node) {
```
Comment (ender336) [inline src/visualization/parsers/boxmodel.ts]: >parent [](start = 34, length = 6)

is this actually optional? looks like domInsert is a noop if it isn't passed

File: src/visualization/parsers/boxmodel.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        if (parent) {
+            if (nextSibling && nextSibling.parentNode === parent) {
+                nextSibling.parentNode.insertBefore(node, nextSibling);
```
Comment (ender336) [inline src/visualization/parsers/boxmodel.ts]: >insertBefore [](start = 39, length = 12)

I think we should have a try-catch that includes this as well as the appendChild on 23

File: src/visualization/parsers/boxmodel.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        }
+        return null;
+    }
+
+    private domRemove(node: Node) {
+        if (node && node.parentNode) {
+            node.parentNode.removeChild(node);
+        }
+        else {
+            console.warn(`Remove: Cannot remove ${node}`);
```
Comment (ender336) [inline src/visualization/parsers/boxmodel.ts]: >console.warn(`Remove: Cannot remove ${node}`); [](start = 12, length = 46)

nit: stylistically we should be consistent with if we are using interpolation

File: src/visualization/parsers/boxmodel.ts (hunk 3 above)
Comment (sarveshnagpal) [reply src/visualization/parsers/boxmodel.ts]: I'll make it required, thanks.

File: src/visualization/parsers/boxmodel.ts (hunk 4 above)
Comment (sarveshnagpal) [reply src/visualization/parsers/boxmodel.ts]: Done, thanks.

File: src/visualization/parsers/boxmodel.ts (hunk 5 above)
Comment (sarveshnagpal) [reply src/visualization/parsers/boxmodel.ts]: Made it consistent.

**Pull request merged**